tests/e2e/test_vault.py

```diff
@@ -0,0 +1,67 @@
+"""
+Module with tests for the use of variables
+"""
+import logging
+import subprocess
+
+import pexpect
+import pytest
+
+from . import utils
+from .settings import SETTINGS
+
+LOGGER = logging.getLogger(__name__)
+DEFAULT_CMD_TIMEOUT = SETTINGS["cmd_timeout"]
+DEFAULT_EVENT_DELAY = SETTINGS["default_event_delay"]
+
+
+@pytest.mark.e2e
+def test_decrypt_vault(update_environment):
+    """
+    Execute a rulebook that has variable encrypted in place or via an
+    external variable.
+    """
+    rulebook = utils.BASE_DATA_PATH / "rulebooks/test_vaulted_rulebook.yml"
+    vars_file = utils.BASE_DATA_PATH / "extra_vars/vaulted_variables.yml"
+    vault_password_file = utils.BASE_DATA_PATH / "passwords/pass3.txt"
+    vault_ids = [
+        ("label1", utils.BASE_DATA_PATH / "passwords/pass1.txt"),
+        (None, utils.BASE_DATA_PATH / "passwords/pass2.txt"),
+    ]
+    cmd = utils.Command(
+        rulebook=rulebook,
+        vars_file=vars_file,
+        vault_ids=vault_ids,
+        vault_password_file=vault_password_file,
+    )
+
+    LOGGER.info(f"Running command: {cmd}")
+    result = subprocess.run(
+        cmd,
+        timeout=DEFAULT_CMD_TIMEOUT,
+        capture_output=True,
+        cwd=utils.BASE_DATA_PATH,
+        text=True,
+    )
+    assert result.returncode == 0
+    assert not result.stderr
+    assert "hello" in result.stdout
+
+
+@pytest.mark.e2e
+def test_decrypt_vault_ask_pass(update_environment):
```

Comment:
you are not using the fixture update_environment in this test. Maybe you forgot to add a test for that?